cvpy/annotation/base/Project.py

```diff
@@ -0,0 +1,129 @@
+from typing import List
+
+from swat.cas import CAS, CASTable
+
+from cvpy.annotation.base.AnnotationLabel import AnnotationLabel
+from cvpy.annotation.base.AnnotationType import AnnotationType
+from cvpy.annotation.base.Credentials import Credentials
+
+
+class Project(object):
+    '''
+    This is a base class for a project in an annotation tool. It has several abstract methods that must be
+    implemented by a subclass.
+    '''
+
+    def __init__(self, cas_session: CAS = None, url: str = None, credentials: Credentials = None,
+                 project_name: str = None, annotation_type: AnnotationType = None, lables: List[AnnotationLabel] = None,
+                 project_id: str = None) -> None:
```

Comment:
What's the difference between project_name and project_id? Can we have just the name as a parameter, and generate an ID inside the class for internal use that the user does not need to see?

Reply:
Good point. I removed project_id from the constructor parameters.